src/javascript/detectors/insecure_object_attribute_modification/insecure_object_attribute_modification.js

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// {fact rule=insecure-object-attribute-modification@v1.0 defects=1}
+var express = require('express')
+var app = express()
+function insecureObjectAttributeModificationNoncompliant() {
+    app.get('www.example.com', (req, res) => {
+        var userId = req.params.id
+        // Noncompliant: external input used as object property.
+        req.session.user[userId] = req.body['userDetails']
+    });
+}
+// {/fact}
+
+
+// {fact rule=insecure-object-attribute-modification@v1.0 defects=0}
+var express = require('express')
+var app = express()
+function insecureObjectAttributeModificationCompliant() {
+    app.get('www.example.com', (req, res) => {
+        var userId = req.params.id
+        // Compliant: Checks whether the property is directly owned by the object before modifying it.
+        if (req.session.user.hasOwnProperty(userId)) {
```

Comment:
From the previous comment, I was asking to check if `userId` is a string type or not. The `hasOwnProperty` check should be applied to `userId` here like -
```
if(!userId.hasOwnProperty("constructor") && !userId.hasOwnProperty("__proto__")) {
     req.session.user[userId] = req.body['userDetails']
}
```

Or since we are checking the type of userId as string, we can directly check datatype using `typeof`.
```
if(typeof userId === 'string'){
     req.session.user[userId] = req.body['userDetails']
}
```